### PR TITLE
TDatD: SSO Industries: Fueling Innovation

### PR DESCRIPTION
**Pull request by jwarwick**

### CODE REVIEW

**Comment by NoahTheDuke** [comment]
Another surprisingly complicated one! Nice. I need to learn about those odd `->>` thingies, obviously.

**Comment by NoahTheDuke** [comment]
Oh cool, quantify is nice!

File: src/clj/game/cards/identities.clj [hunk 1]
```diff
@@ -844,6 +844,44 @@
            :effect (effect (lose :runner :credit 1))
            :msg (msg "make the Runner lose 1 [Credits] by rezzing an Advertisement")}}}
 
+   "SSO Industries: Fueling Innovation"
+   (letfn [(installed-faceup-agendas [state]
+             (->> (all-installed state :corp)
+               (filter #(is-type? % "Agenda"))
+               (filter #(faceup? %))))
```
Comment (nealterrell) [inline src/clj/game/cards/identities.clj]: This can be `(filter faceup?)`

File: src/clj/game/cards/identities.clj [hunk 2]
```diff
@@ -844,6 +844,44 @@
...
+           (selectable-ice? [card]
+             (and
+               (is-type? card "ICE")
+               (installed? card)
+               (zero? (+ (:advance-counter card 0)
+                         (:extra-advance-counter card 0)))))
+           (ice-with-no-advancement-tokens [state]
+             (->> (all-installed state :corp)
+               (filter #(selectable-ice? %))))]
```
Comment (nealterrell) [inline src/clj/game/cards/identities.clj]: Likewise, can be `(filter selectable-ice?)`

File: src/clj/game/cards/identities.clj [hunk 3]
```diff
@@ -844,6 +844,44 @@
...
+     {:events {:corp-turn-ends
+               {:optional
+                {:prompt "Place advancement tokens?"
+                 :req (req (and
+                             (not-empty (installed-faceup-agendas state))
+                             (not-empty (ice-with-no-advancement-tokens state))))
+                 :yes-ability
+                 {:effect (req (show-wait-prompt state :runner "Corp to use SSO Industries' ability")
```
Comment (nealterrell) [inline src/clj/game/cards/identities.clj]: Should be delayed completion.

**Pull request merged**